src/material.cpp

```diff
@@ -476,6 +476,112 @@ void pyne::Material::write_hdf5(std::string filename, std::string datapath,
   delete[] mat_data;
 }
 
+std::string pyne::Material::openmc(std::string frac_type) {
+  std::ostringstream oss;
+
+  pyne::Material temp_mat = this->expand_elements();
+  
+  // vars for consistency
+  std::string new_quote = "\"";
+  std::string end_quote = "\" ";
+  std::string indent = "  ";
+  
+  // open the material element
+  oss << "<material id=" ;
+
+  // add the mat number
+  if(temp_mat.metadata.isMember("mat_number")) {
+    int mat_num = temp_mat.metadata["mat_number"].asInt();
+    oss << new_quote << mat_num << end_quote;
+  }
+  // mat numbers are required for openmc
+  else {
+    oss << new_quote << "?" << end_quote;
+  }
+  // close the material tag
+  oss << ">";
+  // new line
+  oss << std::endl;
+
+  //indent
+  oss << indent;
+
+  // specify density
+  oss << "<density ";
+    // if density is negtaive, report to user
+  if (temp_mat.density < 0.0) {
+    std::cout << "Warning: A density < 0.0 was found. This is not valid for use in OpenMC." << std::endl;
+  }
+  oss << "value=" << std::setprecision(2) << std::fixed << new_quote << temp_mat.density << end_quote;
+  oss << "units=" << new_quote << "g/cc" << end_quote << "/>";
+  // new line
+  oss << std::endl;
+
+  std::map<int, double> fracs;
+  std::string frac_attrib;
+  if(frac_type == "atom") {
+    fracs = temp_mat.to_atom_frac();
+    frac_attrib = "ao=";
+  }
+  else {
+    fracs = temp_mat.comp;
+    frac_attrib = "wo=";
+  }
+  
+  // add nuclides
+  comp_map::iterator f;
+  for(f = fracs.begin(); f != fracs.end(); f++) {
+    //indent
+    oss << "  ";
+    // start a new nuclide element
+    oss << "<nuclide name=";
+    oss << new_quote << pyne::nucname::name(f->first) << end_quote;
```

Comment:
This won't work for metastables. OpenMC expects GND notation which would be, for example, "Am242_m1".

Reply:
Updated the formatting here.

Reply:
Shouldn't this PR include an update to `nucname` to write a nuclide in OpenMC (or apparently GND) format and then just call that?

Reply:
I was thinking that in the car on my way home.. Shouldn't take much to refactor this and make that happen.

Reply:
This meant supporting an `openmc_to_id` function which I knocked out, but now that should probably become part of the `nucname::id` function to support this format more generally. I'll take a look at it tomorrow.